### ISSUE TITLE
Consider switching to TypeScript

### ISSUE THREAD

**Issue by Acconut** [tus/tus-js-client#420]
tus-js-client might benefit from using TypeScript. In particular, the type safety could be helpful if we want to switch to a state-machine. https://transloadit.slack.com/archives/C02J6C6GQKE/p1656938073529209 provides helpful insights for a tool-assisted switch.

**Comment by omBratteng** [comment]
I think this would be a major improvement, as keeping the type definitions file up to date is tedious work and might result in errors.

**Comment by Acconut** [comment]
Yes, I agree. I could see tus-js-client utilizing TS in the near future.
